sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -801,6 +792,8 @@ static int sof_audio_probe(struct platform_device *pdev)
 static const struct platform_device_id board_ids[] = {
 	{
 		.name = "sof_rt5682",
+		.driver_data = (kernel_ulong_t)(SOF_RT5682_MCLK_EN |
+					SOF_SSP_PORT_CODEC(2)),
```

Comment:
SOF_RT5682_MCLK_EN | SOF_SSP_PORT_CODEC(2) is only for BYT and CHT, but "sof_rt5682" is also for ICL. Won't it impact ICL?

Reply:
The DMI quirk should assign the driver_data for ICL.

```
	{
		.callback = sof_rt5682_quirk_cb,
		.matches = {
			DMI_MATCH(DMI_SYS_VENDOR, "Intel Corporation"),
			DMI_MATCH(DMI_PRODUCT_NAME, "Ice Lake Client"),
		},
		.driver_data = (void *)(SOF_RT5682_MCLK_EN |
					SOF_SSP_PORT_CODEC(0)),
	},
```

Reply:
Right, but the quirk is only for ICL RVP. What about other ICL products?

Reply:
```
static unsigned long sof_rt5682_quirk = SOF_RT5682_MCLK_EN |
					SOF_SSP_PORT_CODEC(0);
```

They will use default value in original code. After checking dashboard, ICL program was cancelled. Maybe I just add a icl_rt5682_def board which using SSP0 for codec?

Reply:
> ```
> static unsigned long sof_rt5682_quirk = SOF_RT5682_MCLK_EN |
> 					SOF_SSP_PORT_CODEC(0);
> ```
> 
> They will use default value in original code. After checking dashboard, ICL program was cancelled. Maybe we could:

The default value is
```
static unsigned long sof_rt5682_quirk = SOF_RT5682_MCLK_EN |
                                        SOF_SSP_PORT_CODEC(0);
```
But, after adding ".driver_data = (kernel_ulong_t)(SOF_RT5682_MCLK_EN | SOF_SSP_PORT_CODEC(2))," to the "sof_rt5682" id, sof_rt5682_quirk will be overwritten by sof_audio_probe() for the machines that use "sof_rt5682" machine driver which includes ICL platforms with rt5682 codec.
```
        if (pdev->id_entry && pdev->id_entry->driver_data)
                sof_rt5682_quirk = (unsigned long)pdev->id_entry->driver_data;
```

> 
> 1. Remove DMI quirk of ICL RVP directly.
> 2. Or add a icl_rt5682_def board which is using SSP0 for codec and remove the DMI quirk.
> 

I would choose option 2. And do the corresponding change in snd_soc_acpi_intel_icl_machines[].

> Any idea?

Reply:
done